### DISCUSSION TITLE
Showing date/time the dashboard was last generated

### DISCUSSION THREAD

**Discussion by michalc** [observablehq/framework#1500] [Q&A] [answered]
I'm trying to include the date/time the dashboard was last generated. What's the best way of doing this?

So far I've created a Python data loader to make a 1 row CSV

```python
import datetime
import sys

sys.stdout.buffer.write(b'generated\n')
sys.stdout.buffer.write(str(datetime.datetime.now().timestamp()).encode('utf-8'))
```

And then accessing it in Javascript from a markdown file:

```js
const generated = await FileAttachment("./data/generated.csv").csv({typed: true})
```

And then outputting it in the file, for example:
```html
<p>Data pulled from source ${(new Date(generated[0].generated * 1000)).toLocaleString('en-GB', {dateStyle: 'full', timeStyle: 'long'})}</p>
```
And it works, but it feels a touch manual/faffy. Is there a more concise/canonical way?

**Comment by Fil** [comment, marked as answer]
You can display the build time with a [footer](https://observablehq.com/framework/config#footer) defined as a function that returns something like `built with Observable at ${Date.now()}`. For the data files build time (data loaders might have been run before the site was built, thanks to the cache), use the FileAttachment’s [lastModified](https://observablehq.com/framework/files) property.

**Reply by michalc** [reply]
Thanks! This worked a treat. To tweak the style and timezone I modified it a bit as well:

```
${(new Date()).toLocaleString('en-GB', {dateStyle: 'full', timeStyle: 'long',timeZone: 'Europe/London',})}
```
